src/main/groovy/edu/oregonstate/mist/textbooksapi/TextbooksCollector.groovy

```diff
@@ -0,0 +1,121 @@
+        List<Object> rawTextbooks = objectListCollector(urlString)
+        List<Textbook> textbooks = rawTextbooks.collect { refineTextbook(it) }
+        textbooks
+    }
+
+    /**
+     * Queries Verba compare and builds a list of textbooks for all sections of a course
+     *
+     * @param term
+     * @param department
+     * @param course
+     * @return
+     */
+    static List<Textbook> getTextbooksNoSection(String term, String department, String course) {
+        String urlString = "http://osu.verbacompare.com/compare/courses/?term_id="
+        urlString += "${term}&id=${department}"
+        List<Object> courses = objectListCollector(urlString)
+        Object courseObject = courses.find { it.id == course }
+        List<Textbook> textbooks = []
+        courseObject?.sections?.each { section ->
+            List<Textbook> newBooks = getTextbooks(
+                    term, department, course, section.name
+            )
+            newBooks.each { newBook ->
+                if(!textbooks.find { newBook.id == it.id }) {
+                    textbooks.add(newBook)
+                }
+            }
+        }
+        textbooks
+    }
+
+    /**
+     * Converts verba compare JSON response to Textbook object
+     *
+     * @param rawTextbook
+     * @return
+     */
+    static Textbook refineTextbook(Object rawTextbook) {
+        Float usedPrice = null
+        Float newPrice = null
+        rawTextbook.offers.each {
+            if(it.condition == "new" && it.rental_days == null) {
+                newPrice = Float.parseFloat(it.price)
+            } else if(it.condition == "used" && it.rental_days == null) {
+                usedPrice = Float.parseFloat(it.price)
+            }
+        }
+        Integer bookEdition
+        Integer copyrightYear
+        rawTextbook.with {
+            bookEdition = edition ? Integer.parseInt(edition) : null
+            copyrightYear = copyright_year ? Integer.parseInt(copyright_year) : null
+        }
+
+        new Textbook(
+                id: rawTextbook.isbn,
+                coverImageUrl: rawTextbook.cover_image_url,
+                title: rawTextbook.title,
+                author: rawTextbook.author,
+                edition: bookEdition,
+                copyrightYear: copyrightYear,
+                priceNewUSD: newPrice,
+                priceUsedUSD: usedPrice
+        )
+    }
+
+    /**
+     * Queries Verba compare and returns the JSON list response
+     *
+     * @param urlString
+     * @return
+     */
+    static List<Object> objectListCollector(String urlString) {
+        HttpClient client = HttpClients.createDefault()
+        HttpGet req = new HttpGet(urlString)
+        HttpResponse res = client.execute(req)
+        int status = res.getStatusLine().getStatusCode()
+
+        // Verba compare request should return a 200, regardless of parameters
+        if(status != 200) {
+            throw new Exception("Something went wrong with Verba compare request")
```

Comment:
Could you log a few things here, like the URL that was requested and the response code it returned?